### PR TITLE
Meta: Add a script that prints potential Ladybird cherry-picks

### PR DESCRIPTION
**Pull request by nico**
See comments in the script for details.

Screenshot:

```
% time ./cherry-pick-report.py                                                  
Report of Ladybird commits and their cherry-pick status in SerenityOS:
=====================================================================
[...]
Pull Request: https://github.com/LadybirdBrowser/ladybird/pull/910
  🚫 5f8d852daeae9129d2873b4fce306be3f0599a06 LibWebView+UI: Migrate Ladybird's command line flags to LibWebView

Pull Request: https://github.com/LadybirdBrowser/ladybird/pull/899
  🚫 f735c464d3fe02ac43a1fa46c82ae9a3bb5de8b1 LibWeb/Fetch: Don’t cache redirects
[...]
=====================================================================
1440 PRs, 407 cherry-picked PRs, 113 unwanted PRs
920 to go
=====================================================================
```

(The first of those should be added to `never_merge_prs` probably, and the second should probably be cherry-picked, but maybe it depends on other earlier PRs.)

@circl-lastname

### CODE REVIEW

**Comment by circl-lastname** [comment]
Whoa, Very cool!

File: Meta/lb-cherry-picks.py [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+# Define a list of pull request IDs that should never be merged
+# and the reasons why they shouldn't be merged.
+# We fairly likely don't want the PRs in here (but it isn't 100% set
+# in stone for some of them). But if a PR is not in here, it doesn't
+# necessarily mean that we want it: It might just not be triaged yet.
+# If we want some commits of a PR but not all, the PR should not be
+# in this list -- but some commits of the PR we don't want in
+# never_merge_commits further down instead.
+never_merge_prs = {
+    "https://github.com/LadybirdBrowser/ladybird/pull/16": "Ladybird-specific",
```
Comment (circl-lastname) [inline Meta/lb-cherry-picks.py:259]: What about PRs 1-15?

**Comment by circl-lastname** [comment]
There's also a list of [non PR commits](https://github.com/LadybirdBrowser/ladybird/activity?ref=master&activity_type=direct_push) to keep in mind

File: Meta/lb-cherry-picks.py (hunk 1 above)
Comment (nico) [reply Meta/lb-cherry-picks.py:261]: Good question! PR15 is actually here, two lines down. I sorted these by merge time, not by PR number, at the start. That's a pain to maintain though, so I moved it to sorted now.

What about PRs 1-14? It looks like the merged commits there went to a different branch for some reason, and they show up as `Commit: https://github.com/LadybirdBrowser/ancient-history/commit/dd1fbd3513` and don't have a `Pull-request:` footer. I annotated commits in `never_merge_commits` with their PR number.

**Comment by nico** [comment]
Thanks for taking a look!

> There's also a list of [non PR commits](https://github.com/LadybirdBrowser/ladybird/activity?ref=master&activity_type=direct_push) to keep in mind, usually emergency build fixes, but sometimes full changes.

That's handled by

```
        # The previous commit had no associated pull request.
        if current_commit and not current_pr:
            ladybird_commits.setdefault(None, []).append(current_commit)
```

around line 69. Commits without PR are assigned the PR `None`.

(Technically, we'd have to have the same check after the loop, else we miss the very first commit. That's the commit belonging to PR1, and we don't want it, and having two extra lines to know about it, and then another to annotate that we don't want it doesn't seem super useful maybe.)

File: Meta/lb-cherry-picks.py (hunk 1 above)
Comment (circl-lastname) [reply Meta/lb-cherry-picks.py:261]: Oh maybe at some point they were trying to revive that repo? Well as long as those don't show up then it's ok

**Pull request merged**